cluster-up/cluster/kind-k8s-sriov-1.17.0/certcreator/certsecret.go

```diff
@@ -0,0 +1,139 @@
+package main
+
+import (
+	"encoding/base64"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"strings"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+
+	"kubevirt.io/kubevirtci/cluster-up/cluster/kind-k8s-sriov-1.17.0/certcreator/certlib"
+)
+
+func handleKubeClientConfig(kubeconfig string) (*rest.Config, error) {
+	if kubeconfig == "" {
+		log.Printf("Using env kubeconfig %s", kubeconfig)
+		kubeconfig = os.Getenv("KUBECONFIG")
+	}
+
+	var config *rest.Config
+	var err error
+	if kubeconfig != "" {
+		log.Printf("Loading kube client config from path %q", kubeconfig)
+		config, err = clientcmd.BuildConfigFromFlags("", kubeconfig)
+	} else {
+		log.Printf("Using in-cluster kube client config")
+		config, err = rest.InClusterConfig()
+	}
+	if err != nil {
+		return nil, fmt.Errorf("could not get the client: %v", err)
+	}
```

Comment:
This logic repeats a lot,
since it exit(1), it is possible to wrap it in a function and use all over please

Reply:
I am not sure what you expect to be warped, could you please give an example?

Reply:
Refereed mainly to the ones that use FatalF which exit(1).
But in case there are others which also can be replaced with FatalF this function can be used:
```
func exitOnError(err, string) 
{
   if err != nil ... return fmt.FatalF
}
```

Reply:
log.Fatal is being used only on main function because it logs the error and return 1
Thanks for explaining, I am not sure how it improves the code I mean its the nature of Go to get an error and immediately handle it 
lets hear other reviewer opinions.

Reply:
it will be one liner instead 3 , in the N appearances in main

Reply:
@oshoval I think you have commented wrong part of the code, this comments belongs to https://github.com/kubevirt/kubevirtci/pull/472/files#diff-af1a757576e17f47e55b5484a3ca9ff3fc9c77f5b097c156faa77ba06760900fR122-R140 right ?

Reply:
Yep
(see please first line https://github.com/kubevirt/kubevirtci/pull/472#discussion_r521196259, which says that i ref the wrong one as you said)